### ISSUE TITLE
com.hazelcast.jet.core.MemberReconnectionStressTest.test

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#22649]
_master_ (commit 470f95b02543901406fa5e3b232545b5faf8b7c8)

Failed on ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8-nightly/4/testReport/com.hazelcast.jet.core/MemberReconnectionStressTest/test/

<details><summary>Stacktrace:</summary>

```
com.hazelcast.jet.core.JobNotFoundException: Job with id 08df-340e-bca2-0001 not found
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:924)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:807)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
	at ------ submitted from ------.()
	at com.hazelcast.internal.util.ExceptionUtil.cloneExceptionWithFixedAsyncStackTrace(ExceptionUtil.java:348)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.returnOrThrowWithGetConventions(InvocationFuture.java:112)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:100)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:617)
	at com.hazelcast.jet.impl.JobProxy.getStatus0(JobProxy.java:72)
	at com.hazelcast.jet.impl.AbstractJobProxy.getStatus(AbstractJobProxy.java:177)
	at com.hazelcast.jet.core.JetTestSupport.lambda$ditchJob$6(JetTestSupport.java:410)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1338)
	at com.hazelcast.jet.core.JetTestSupport.ditchJob(JetTestSupport.java:409)
	at com.hazelcast.jet.core.JetTestSupport.shutdownJobsAndGetLeakedClassLoaders(JetTestSupport.java:134)
	at com.hazelcast.jet.core.JetTestSupport.shutdownFactory(JetTestSupport.java:106)
	at sun.reflect.GeneratedMethodAccessor373.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:807)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-ZingJDK8-nightly/4/Hazelcast-master-ZingJDK8-nightly-4.zip

**Comment by AyberkSorgun** [comment]
Internal JIRA issue: [HZ-1689](https://hazelcast.atlassian.net/browse/HZ-1689)

**Comment by sumnerib** [comment]
Failed again on _master_ with IbmJDK8-fips: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips-nightly/477/testReport/junit/com.hazelcast.jet.core/MemberReconnectionStressTest/test/

**Comment by olukas** [comment]
Failed again on _master_ with IbmJDK8-fips: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8-fips-nightly/478/testReport/com.hazelcast.jet.core/MemberReconnectionStressTest/test/

**Comment by sumnerib** [comment]
Failed again on _5.2.z_ with ZuluJDK17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-ZuluJDK17-nightly/38/testReport/junit/com.hazelcast.jet.core/MemberReconnectionStressTest/test/

**Comment by olukas** [comment]
Failed again on _5.2.z_ with IbmJDK8-nightly: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-IbmJDK8-nightly/59/testReport/com.hazelcast.jet.core/MemberReconnectionStressTest/test/

**Comment by olukas** [comment]
@orcunc could you please back-port the fix to 5.2.z branch?
